packages/rum/src/boot/startRecording.ts

```diff
@@ -13,17 +13,20 @@ export function startRecording(
   session: RumSession,
   parentContexts: ParentContexts
 ) {
-  const { addRecord, stop: stopSegmentCollection } = startSegmentCollection(
+  const segmentCollection = startSegmentCollection(
     lifeCycle,
     applicationId,
     session,
     parentContexts,
     (data, meta, rawSegmentSize, flushReason) =>
       send(configuration.sessionReplayEndpoint, data, meta, rawSegmentSize, flushReason)
   )
+  if (!segmentCollection) {
+    return
+  }
 
   function addRawRecord(rawRecord: RawRecord) {
-    addRecord({ ...rawRecord, timestamp: Date.now() })
+    segmentCollection!.addRecord({ ...rawRecord, timestamp: Date.now() })
```

Comment:
Having `!` is not great, do we have a unit test where we return undefined or null to avoid crashing at runtime?

Reply:
I removed the `!` and added a test.